### PR TITLE
Enhance SetupConnection Handling and Implement Protocol-Specific Flag Checks

### PR DESCRIPTION
**Pull request by xyephy**
This PR works on addressing the discussions in issue #853

### CODE REVIEW

File: protocols/v2/subprotocols/common-messages/src/setup_connection.rs [hunk 1]
```diff
@@ -389,13 +400,22 @@ mod test {
     use core::convert::TryInto;
 
     #[test]
-    fn test_check_flag() {
-        let protocol = crate::Protocol::MiningProtocol;
-        let flag_avaiable = 0b_0000_0000_0000_0000_0000_0000_0000_0000;
+    fn test_check_flags() {
+        let protocol = Protocol::MiningProtocol;
+        let flag_available = 0b_0000_0000_0000_0000_0000_0000_0000_0001;
```
Comment (Fi3) [inline protocols/v2/subprotocols/common-messages/src/setup_connection.rs:405]: why you changed this test?

Comment (xyephy) [reply protocols/v2/subprotocols/common-messages/src/setup_connection.rs:405]: thought of it as a way ensuring that check_flags works when both available and required flags are the same for MiningProtocol and JobDeclarationProtocol.

Comment (xyephy) [reply protocols/v2/subprotocols/common-messages/src/setup_connection.rs:405]: @Fi3 when I did the revert I keep getting this error:
```
test setup_connection::test::test_check_flag ... FAILED
test setup_connection::test::test_has_work_selection ... ok
test setup_connection::test::test_set_requires_std_job ... ok

failures:

---- setup_connection::test::test_check_flag stdout ----
thread 'setup_connection::test::test_check_flag' panicked at v2/subprotocols/common-messages/src/setup_connection.rs:408:9:
assertion failed: SetupConnection::check_flags(protocol, flag_avaiable, flag_required)
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
    setup_connection::test::test_check_flag
    
    ```
    
    any idea on how I can go about it?

Comment (xyephy) [reply protocols/v2/subprotocols/common-messages/src/setup_connection.rs]: @Fi3 I managed to fix the error, I think you can review again and give feedback

**Comment by plebhash** [comment]
@xyephy looks like Rustfmt CI is broken, can you do `cargo +nightly fmt` on all crate workspaces, commit and push the changes, please?

**Comment by plebhash** [comment]
@xyephy did you explore the MG tests? I'm wondering if this behavior is already covered in our MG suite?

**Comment by xyephy** [comment]
> @xyephy looks like Rustfmt CI is broken, can you do `cargo +nightly fmt` on all crate workspaces, commit and push the changes, please?

Thanks @plebhash, let me work on that.

**Comment by xyephy** [comment]
> @xyephy did you explore the MG tests? I'm wondering if this behavior is already covered in our MG suite, or should we add a new one?

for this issue I hadn't done that, let me explore/review then give you feedback.

**Comment by plebhash** [comment]
@xyephy FYI I merged https://github.com/stratum-mining/stratum/pull/945 but apparently that introduced conflics to your PR

**Comment by xyephy** [comment]
> @xyephy FYI I merged #945 but apparently that introduced conflics to this PR

I will do the cleanups to solve the conflicts.

**Comment by plebhash** [comment]
@xyephy any progress on writing the MG tests for this one?

**Comment by xyephy** [comment]
> @xyephy any progress on writing the MG tests for this one?

I'm 80% done on this

**Comment by plebhash** [comment]
closing in favor of #1035

**Pull request closed without merging**